test/unit/mdc-list/mdc-list.test.js

```diff
@@ -72,6 +78,43 @@ test('#adapter.getListItemIndex returns the index of the element specified', ()
   document.body.removeChild(root);
 });
 
+test('#adapter.setAttributeForElementIndex sets the attribute for the list element at index specified', () => {
+  const {root, component} = setupTest();
+  document.body.appendChild(root);
+  const selectedNode = root.querySelectorAll('.mdc-list-item')[1];
+  component.getDefaultFoundation().adapter_.setAttributeForElementIndex(1, 'foo', 'bar');
+  assert.equal('bar', selectedNode.getAttribute('foo'));
+  document.body.removeChild(root);
+});
+
+test('#adapter.removeAttributeForElementIndex sets the attribute for the list element at index specified', () => {
+  const {root, component} = setupTest();
+  document.body.appendChild(root);
+  const selectedNode = root.querySelectorAll('.mdc-list-item')[1];
+  component.getDefaultFoundation().adapter_.setAttributeForElementIndex(1, 'foo', 'bar');
+  component.getDefaultFoundation().adapter_.removeAttributeForElementIndex(1, 'foo');
+  assert.isFalse(selectedNode.hasAttribute('foo'));
+  document.body.removeChild(root);
+});
+
+test('#adapter.addClassForElementIndex adds the class to the list element at index specified', () => {
+  const {root, component} = setupTest();
+  document.body.appendChild(root);
+  const selectedNode = root.querySelectorAll('.mdc-list-item')[1];
+  component.getDefaultFoundation().adapter_.addClassForElementIndex(1, 'foo');
+  assert.isTrue(selectedNode.classList.contains('foo'));
+  document.body.removeChild(root);
+});
+
+test('#adapter.removeClassForElementIndex removes the class from the list element at index specified', () => {
+  const {root, component} = setupTest();
+  document.body.appendChild(root);
+  const selectedNode = root.querySelectorAll('.mdc-list-item')[1];
+  component.getDefaultFoundation().adapter_.removeClassForElementIndex(1, 'foo');
+  assert.isFalse(selectedNode.classList.contains('foo'));
```

Comment:
when does the element get the foo class? I ran this and it looks like if you remove line 113 it still passes